Microsoft.DotNet.Interactive.FSharp/FSharpKernel.fs

```diff
@@ -137,21 +136,8 @@ type FSharpKernel() as this =
             context.Publish(CompletionRequestCompleted(completionItems, requestCompletion))
         }
 
-    let mutable _assemblyProbingPaths = Unchecked.defaultof<AssemblyResolutionProbe>
-    let mutable _nativeProbingRoots = Unchecked.defaultof<NativeResolutionProbe>
-
-    let dependencies =
-        let createDependencyProvider () =
-            // These may not be set to null, if they are it is a product coding error
-            // ISupportNuget.Initialize must be invoked prior to creating the DependencyManager
-            // With non null funcs
-            if isNull _assemblyProbingPaths then
-                raise (new InvalidOperationException("_assemblyProbingPaths is null"))
-
-            if isNull _nativeProbingRoots then
-                raise (new InvalidOperationException("_nativeProbingRoots is null"))
-            new DependencyProvider(_assemblyProbingPaths, _nativeProbingRoots)
-        lazy (createDependencyProvider ())
+    let _packageRestoreContext = lazy (new PackageRestoreContext())
+    do base.RegisterForDisposal(fun () -> if _packageRestoreContext.IsValueCreated then _packageRestoreContext.Value.Dispose())
```

Comment:
Same as with `CSharpKernel`: You can do the registration inside the `Lazy` setup.